mldsa/polyvec.c

```diff
@@ -201,16 +201,37 @@ int polyvecl_chknorm(const polyvecl *v, int32_t bound)
 /**************************************************************/
 
 void polyveck_uniform_eta(polyveck *v, const uint8_t seed[MLDSA_CRHBYTES],
-                          uint16_t nonce)
+                          const uint16_t nonce)
 {
   unsigned int i;
-
+  uint16_t n = nonce;
   for (i = 0; i < MLDSA_K; ++i)
+  __loop__(
+    assigns(i, n, object_whole(v))
+    invariant(i <= MLDSA_K)
+    invariant(forall(k1, 0, i,
+      array_abs_bound(v->vec[k1].coeffs, 0, MLDSA_N, MLDSA_ETA + 1))))
   {
-    poly_uniform_eta(&v->vec[i], seed, nonce++);
+    poly t;
+    poly_uniform_eta(&t, seed, n);
+
+#ifdef CBMC
+#pragma CPROVER check push
+#pragma CPROVER check disable "unsigned-overflow"
+#endif
+    n++;
+#ifdef CBMC
+#pragma CPROVER check pop
+#endif
```

Comment:
The function is only called with nonce=L`, so this can never overflow in practise. I'd prefer a stronger pre-condition over silencing the warning here.